app/src/main/java/org/p2p/wallet/user/interactor/UserTokensInteractor.kt

```diff
@@ -72,14 +126,28 @@ class UserTokensInteractor(
     suspend fun findUserToken(mintAddress: String): Token.Active? =
         userTokensLocalRepository.getUserTokens().find { it.mintAddress == mintAddress }
 
-    fun observeUserBalance(): Flow<BigDecimal> {
-        return userTokensLocalRepository.observeUserBalance()
-    }
-
-    suspend fun getUserBalance(): BigDecimal {
-        return userTokensLocalRepository.getUserBalance()
-    }
-
     suspend fun setTokenHidden(mintAddress: String, visibility: String) =
         userTokensLocalRepository.setTokenHidden(mintAddress, visibility)
+
+    private suspend fun createNewToken(
+        mintAddress: Base58String,
+        newBalanceLamports: BigInteger,
+        accountPublicKey: Base58String
+    ): Token.Active? {
+
+        val tokenMetadataData = userLocalRepository.findTokenData(mintAddress.base58Value) ?: return null
```

Comment:
can we have a task to rename UserLocalRepository?

Reply:
https://p2pvalidator.atlassian.net/browse/PWN-9649
Added todo inside the repo